src/node_report.cc

```diff
@@ -0,0 +1,790 @@
+      char pathname[NR_MAXPATH + NR_MAXNAME + 1] = "";
+      strcpy(pathname, report_directory);
+#ifdef _WIN32
+      strcat(pathname, "\\");
+#else
+      strcat(pathname, "/");
+#endif
+      strncat(pathname, filename, NR_MAXNAME);
+      fp = fopen(pathname, "w");
+    } else {
+      fp = fopen(filename, "w");
+    }
+    // Check for errors on the file open
+    if (fp == NULL) {
+      if (strlen(report_directory) > 0) {
+        fprintf(stderr, "\nFailed to open Node.js report file: %s directory: %s (errno: %d)\n", filename, report_directory, errno);
+      } else {
+        fprintf(stderr, "\nFailed to open Node.js report file: %s (errno: %d)\n", filename, errno);
+      }
+      return;
+    } else {
+      fprintf(stderr, "\nWriting Node.js report to file: %s\n", filename);
+    }
+  }
+
+  // Print NodeReport title and event information
+  fprintf(fp, "================================================================================\n");
+  fprintf(fp, "==== NodeReport ================================================================\n");
+
+  fprintf(fp, "\nEvent: %s, location: \"%s\"\n", message, location);
+  fprintf(fp, "Filename: %s\n", filename);
+
+  // Print dump event and module load date/time stamps
+#ifdef _WIN32
+  fprintf(fp, "Dump event time:  %4d/%02d/%02d %02d:%02d:%02d\n",
+          tm_struct.wYear, tm_struct.wMonth, tm_struct.wDay,
+          tm_struct.wHour, tm_struct.wMinute, tm_struct.wSecond);
+  fprintf(fp, "Module load time: %4d/%02d/%02d %02d:%02d:%02d\n",
+          loadtime_tm_struct.wYear, loadtime_tm_struct.wMonth, loadtime_tm_struct.wDay,
+		  loadtime_tm_struct.wHour, loadtime_tm_struct.wMinute, loadtime_tm_struct.wSecond);
+#else  // UNIX, OSX
+  fprintf(fp, "Dump event time:  %4d/%02d/%02d %02d:%02d:%02d\n",
+          tm_struct.tm_year+1900, tm_struct.tm_mon+1, tm_struct.tm_mday,
+          tm_struct.tm_hour, tm_struct.tm_min, tm_struct.tm_sec);
+  fprintf(fp, "Module load time: %4d/%02d/%02d %02d:%02d:%02d\n",
+		  loadtime_tm_struct.tm_year+1900, loadtime_tm_struct.tm_mon+1, loadtime_tm_struct.tm_mday,
+		  loadtime_tm_struct.tm_hour, loadtime_tm_struct.tm_min, loadtime_tm_struct.tm_sec);
+#endif
+
+  // Print Node.js and deps component versions
+  fprintf(fp, "\nNode.js version: %s\n", NODE_VERSION);
+  fprintf(fp, "(v8: %s, libuv: %s, zlib: %s, ares: %s)\n",
+        V8::GetVersion(), uv_version_string(), ZLIB_VERSION, ARES_VERSION_STR);
+
+  // Print OS name and level and machine name
+#ifdef _WIN32
+  fprintf(fp,"\nOS version: Windows ");
+#if defined(_MSC_VER) && (_MSC_VER >= 1900)
+  if (IsWindows1OrGreater()) {
+    fprintf(fp,"10 ");
+  } else
+#endif
+  if (IsWindows8OrGreater()) {
+    fprintf(fp,"8 ");
+  } else if (IsWindows7OrGreater()) {
+    fprintf(fp,"7 ");
+  } else if (IsWindowsXPOrGreater()) {
+    fprintf(fp,"XP ");
+  }
+  if (IsWindowsServer()) {
+    fprintf(fp,"Server\n");
+  } else {
+    fprintf(fp,"Client\n");
+  }
+  TCHAR  infoBuf[256];
+  DWORD  bufCharCount = 256;
```

Comment:
One space too many between the type and the variable name.  (Also, it's somewhat inconsistent to use camelCase here and snake_case everywhere else.)